src/template.c

```diff
@@ -46,7 +62,9 @@ std::vector<std::string> template_dfa(CharacterVector x) {
       switch (c) {
       case '{':
         state = code;
-        pieces.push_back(input.substr(pieceStartIdx, i - pieceStartIdx - 1));
+        str = PROTECT(Rf_mkCharLenCE(input + pieceStartIdx, i - pieceStartIdx - 1, CE_UTF8));
```

Comment:
This assumes the input is UTF-8, if that is not a good assumption we can change this to just be `Rf_mkCharLen()`, which will use the native encoding

Reply:
The calling function does convert to UTF-8, but this function should probably throw a warning if the input is not UTF-8, just in case someone else wants to call it.

Reply:
Detecting this is slightly more complicated that just checking the return of `Rf_getCharCE()` as you also have to check if the string is all ASCII (which have a CE_NATIVE encoding), but I have now added this warning.